### ISSUE TITLE
[install bug] Python path !

### ISSUE THREAD

**Issue by yet-another-average-joe** [akaJes/marlin-config#59]
Hello akaJes, and first of all many thanks for this amazing piece of software ! It should come with Marlin firmware !

When I first started with marlin-conf, I did'nt find much information on the web...
So now, I am currently making video tutorials about marlin-conf, english and french. At the moment, 50% done. I had to explore marlin-conf extensively, and discovered some glitches.

First one pops up while installing PlatformIO Core from the marlin-conf bootscreen.

OS : Windows 7
Pyhton not installed
PlatfomIO not installed :
Fresh install for evrything, tested on real machine and VirtualBox.

PlatformIO button installs PlatformIO and Python 2.7. But when restarting marlin-conf, PlatformIO appears as not installed. In fact it is, and the problem is Python 2.7

The problem is extremely easy to solve. Python does not include itself in the PATH;

2 fixes for the user :

fix 1 : manually add Python to the PATH :

in user PATH or system PATH, add :

C:\Users\<USER>\.platformio\python27\;C:\Users\<USER>\.platformio\python27\Scripts;C:\Users\<USER>\.platformio\python27\Tools

fix 2 :

reinstall Python, either as standalone after a download from Python website, or (prefered IMHO) from : 
C:\Users\admin\.platformio\.cache\python-2.7.14.amd64.msi and then choose "change Python 2.7", *NOT* "Repair"
Then, enable "Add python.exe to Path" -> Will be installed on local hard drive

And marlin-conf will be happy !

(I don't know much about Python install options, but I feel like it is just a forgotten option in the silent install command line)

If you'd like I can upload a short vid to Youtube, showing the problem/solution

(I can upload a quick video on YouTube if you want)
(sorry for my bad english)

**Comment by akaJes** [comment]
I'm tried it out on Windows10 and windows2008 without any issues width latest version https://github.com/akaJes/marlin-config/releases/tag/v2.10.3
The `C:\Users<USER>.platformio` folder automaticaly added when the application starts

**Comment by yet-another-average-joe** [comment]
There seems there's an issue with python 2.7 windows msi installer. I found this : 

https://stackoverflow.com/questions/34214607/python-2-7-windows-silent-installer-msi-command-line-option-to-set-the-path

I will give it a try with win8 and win10 ASAP

**Comment by akaJes** [comment]
I do not use standalones installers. Everything bundled inside one automated installer (Python, libs, platformIO) and placed in `C:\Users<USER>.platformio` folder

**Comment by madmax2** [comment]
@yet-another-average-joe youtube video is not available
please fix

**Comment by yet-another-average-joe** [comment]
Sorry, but this video does not exist anymore (removed from YouTube, and deleted from the HDD). It was intented for the developper (akajes).
It was just showing the problem and the 2 fixes : just follow the steps in post #1

**Comment by madmax2** [comment]
@yet-another-average-joe is PlatformIO needed to be installed to compile the marlin fw?

Do we still need to do fix 1 in order for marlin conf to be happy and compile the fw or will it still compile regardless if we do the fix 1 or not since as you said PlatformIO is actually installed already, but it is a Python 2.7 issue..?

Also can you explain further, why marlin conf sees it is not installed due to the Python 2.7 issue?

**Comment by yet-another-average-joe** [comment]
I feel like you are not fluent with Microsoft OS's. Old players know what PATH is (it's been around for the last 35 years, since PC-DOS and MS-DOS 2.0, and it had to be configured by hand on each and every PC up to Windows 98).  So you'd better to follow fix 2 instructions. You will not have to edit the PATH environment variable. The msi file will do it for you !

**Comment by madmax2** [comment]
Don't assume/jump to conclusion that I am not not fluent with an OS just cos I ask those question..
I have been using Windows Operating systems for a long time, 
but there is always something new to learn daily..

I am just trying to understand the underlying issue of this marlin conf and platformIO showing it is not installed.

And yes I know what PATH is..

I already did Fix 1 before I asked you about the youtube video,
but I am just trying understand the issue here, in regards to marlin conf, platformIO and python 2.7.

BTW, anyone who is posting on github, would likely to have some sort of technical knowledge already
so something like editing a PATH is not something they would not know about..
The average joe would not even be posting on github.

===

If you start assuming stuff, then by the same logic, anyone using 
marlin conf is a total moron that does not know how to edit a configuration.h file in notepad++ 

So please don't assume/jump to conclusions so fast in the future.

**Comment by yet-another-average-joe** [comment]
If you know what PATH is, the answer will be obvious !

marlin-conf thinks that platformio is not installed because platformio relies on Python. Python not being in the PATH, it cannot access Python and therefore platformio. So it assumes it is not installed. 

And I am sorry to try to help you, and sorry to give 2 fixes. I will stop there.

**Comment by madmax2** [comment]
No, just cos it is obvious to you
does not mean everyone gets it as quickly as you do..
This is a totally new software some people are just learning and they are just trying to understand what is happening..

Just cos you have driven a car for 20 years, don't mean you can just jump in a truck and be an expert truck driver. 

Instead of posting your harsh, rude response and irrelevant to what I was asking...
and taken this thread off topic
You could've just posted that answer to what I was asking
and anyone else who had googled this problem could have read it and understood the cause.
And thread would have been finished..

So don't assume stuff so quickly in the future
It just makes an ASS out of U and ME

**Comment by yet-another-average-joe** [comment]
There are only 2 harsh posts there : yours, and you did'nt even let know if the fixes worked for you !

Sorry, I do'nt know what this mean : "It just makes an ASS out of U and ME". Sorry again, english is not my first langage, and I did'nt learn argotic vocabulary..

**Comment by madmax2** [comment]
No, you are the one that started it, by make assumptions that I don't know anything about Windows OS
and posting a useless comment that had nothing to do with what I was asking..

**"I feel like you are not fluent with Microsoft OS's. Old players know what PATH is (it's been around for the last 35 years, since PC-DOS and MS-DOS 2.0, and it had to be configured by hand on each and every PC up to Windows 98). "**

Did I ask anything about what PATH is?
Go read my questions again,

I don't need to post if I I did the fix or not
since it is very likely that I did do the fix already, if I had read your first post and this entire thread already..
It is not that hard to add some path locations..
and even if I had been a total noob about PATH, I could have googled how add a PATH in Windows

I was trying to understand more, since I am still new to marlin conf and learning..

"It just makes an ASS out of U and ME"
Why don't you google it...
The first few links will tell you everything about that statement.

**Comment by GhostlyCrowd** [comment]
would you two shut the fuck up already some people are subscribed to this repository for updates on constructive things not childish bickering.
